content/en/developers/integrations/log_pipeline.md

```diff
@@ -16,111 +16,165 @@ further_reading:
 
-To add a facet or measure:
+Not all attributes are meant to be used as a facet. The need for facets in integrations is focused on two things:
+* Facets provide a straightforward interface for filtering logs. They are leveraged in Log Management autocomplete features, allowing users to find and aggregate key information found in their logs.
+* Facets allow for attributes with low readability to be renamed with a label that is easier to understand. For example: `@deviceCPUper` → `Device CPU Utilization Percentage`.
+
+You can create [facets][12] in the [Log Explorer][16].
+
+#### Create facets
+
+Correctly defining facets is important as they improve the usability of logs in analytics, monitors, and aggregation features across Datadog's logs management product. 
+
+They allow for better findability of application logs by populating autocomplete features across Log Management.
+
+<div class="alert alert-info">Quantitative facets, called "Measures", allow users to filter logs over a range of numeric values using relational operators.  
+For example, a measure for a latency attribute allows users to search for all logs greater-than a certain duration. </div>
+
+##### Requirements
+
+Attributes mapped to custom facets must be namespaced first
+: Generic custom attributes that do not map to [Datadog Standard Attribute][6] must be namespaced when used with custom [facets][14]. An [Attribute Remapper][5] can be used to namespace an attribute with the integration's name.  
+For example, remapping `attribute_name` to `integration_name.attribute_name`.
+
+Custom facets must not duplicate an existing Datadog Facet
+: To avoid confusion with existing out-of-the-box Datadog facets, avoid creating custom facets that duplicate an existing facets mapped to [Datadog Standard Attributes][6].
+
+Custom facets must be grouped under the `source` name
+: When creating a custom facet a group should be assigned. Set the `Group` value to the `source`, same as the integration's name.
+
+Custom facets must have the same data type as the mapped attribute
+: Set the facet data type (String, Boolean, Double, or Integer) to the same type as the Attribute mapped to it. Mismatched types prevent the facet from being used as intended and can cause it to populate incorrectly.
+
+**Add a facet or measure**
 
 1. Click on a log that contains the attribute you want to add a facet or measure for. 
 2. In the log panel, click the Cog icon next to the attribute.
 3. Select **Create facet/measure for @attribute**.
 4. For a measure, to define the unit, click **Advanced options**. Select the unit based on what the attribute represents.
-5. Click **Add**.
+**Note**: Define the [unit][11] of a measure based on what the attribute represents.
+5. To help navigate the facet list specify a facet **Group**. If one does not exist enter the name of the group matching the source tag and a description of the new group, and select **New group**.
```

Comment:
add a comma after list or change sentence to "Specify a facet group to help navigate the facet list"

Reply:
and then "If the facet group does not exist, enter the name"...